Classes/Form/FormElement/FileCollectionElement.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * Copyright (C) 2023 Daniel Siepmann <coding@daniel-siepmann.de>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+namespace WerkraumMedia\FormFileCollection\Form\FormElement;
+
+use TYPO3\CMS\Core\Resource\FileCollectionRepository;
+use TYPO3\CMS\Core\Resource\FileInterface;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Form\Domain\Model\FormElements\AbstractFormElement;
+
+/**
+ * Elements are created with constructor arguments and don't have DI available.
+ */
+final class FileCollectionElement extends AbstractFormElement
+{
+    public function setProperty(string $key, $value): void
+    {
+        if ($key === 'fileCollection' && is_array($value)) {
+            $this->setProperty('options', $this->getOptions($value));
+            return;
+        }
+
+        parent::setProperty($key, $value);
+    }
+
+    /**
+     * @param array<string, string> $configuration
+     * @return array<string, string>
```

Comment:
```suggestion

     * @return array<string, string>
```

It's like this in line 76-78.

Reply:
Added cs fixer rule and applied it.